src/c-api.cpp

```diff
@@ -111,6 +117,125 @@ const qljs_web_demo_diagnostic* qljs_web_demo_lint(qljs_web_demo_parser* p) {
   return p->lint();
 }
 
+#if QLJS_SUBLIME_TEXT_PLUGIN
+struct qljs_sublime_text_3_parser final
+    : public quick_lint_js::qljs_parser_base<
+          quick_lint_js::sublime_text_3_locator,
+          quick_lint_js::c_api_error_reporter<
+              qljs_sublime_text_3_diagnostic,
+              quick_lint_js::sublime_text_3_locator>> {
+ public:
+  void set_text(quick_lint_js::string8_view replacement) {
+    this->document_.set_text(replacement);
+  }
+};
+
+qljs_sublime_text_3_parser* qljs_sublime_text_3_create_parser(void) {
+  QLJS_SUBLIME_TEXT_DEFINE_SIGNAL_HANDLER();
+  qljs_sublime_text_3_parser* p = new qljs_sublime_text_3_parser();
+  return p;
+}
+
+void qljs_sublime_text_3_destroy_parser(qljs_sublime_text_3_parser* p) {
+  delete p;
+}
+
+const qljs_sublime_text_3_error* qljs_sublime_text_3_set_text(
+    qljs_sublime_text_3_parser* p, const void* text_utf_8,
+    size_t text_byte_count) {
+  QLJS_SUBLIME_TEXT_TRY() {
+    p->set_text(quick_lint_js::string8_view(
+        reinterpret_cast<const quick_lint_js::char8*>(text_utf_8),
+        text_byte_count));
+    return nullptr;
+  }
+  QLJS_SUBLIME_TEXT_CATCH() {
+    const char* message = QLJS_SUBLIME_TEXT_COPY_PROGRAM_REPORT();
+    QLJS_SUBLIME_TEXT_CLEAR_PROGRAM_REPORT();
+    return new qljs_sublime_text_3_error{message};
+  }
+}
+
+const qljs_sublime_text_3_result* qljs_sublime_text_3_lint(
+    qljs_sublime_text_3_parser* p) {
+  QLJS_SUBLIME_TEXT_TRY() {
```

Comment:
Must fix: quick-lint-js's parser already has opt-in crash resilience. The LSP server uses this to mitigate crashes. We just need to enable it for the Sublime Text plugin. See lines 34-36 in this file. (The function is now named `parse_and_visit_module_catching_fatal_parse_errors`. Commit fed9d4977f1a42cd17ce501bf3d76609b902adf2 fixed the comment, but your work is based on an older commit.)

If we use `parse_and_visit_module_catching_fatal_parse_errors`, I think we no longer need the program report stuff. (I'm fine with the plugin crashing on `QLJS_ASSERT` failures.)

Reply:
`parse_and_visit_module_catching_fatal_parse_errors` will show unexpected-token errors as normal diagnostics. This means we don't need custom code for a pop-up (at least for unexpected-token errors). I think a squiggly is a better UI than a pop-up anyway. (A pop-up is very distracting.)